src/Sulu/Bundle/ContentBundle/Resources/public/js/ckeditor/internal-link.js

```diff
@@ -44,29 +45,74 @@ define([
             tagName: 'sulu:link',
 
             init: function(editor) {
-                // extend dtd
-                CKEDITOR.dtd[this.tagName] = 1;
-                CKEDITOR.dtd.body['sulu:link'] = 1;
-                CKEDITOR.dtd.div['sulu:link'] = 1;
-                CKEDITOR.dtd.li['sulu:link'] = 1;
-                CKEDITOR.dtd.p['sulu:link'] = 1;
-                CKEDITOR.dtd.$block['sulu:link'] = 1;
-                CKEDITOR.dtd.$removeEmpty['sulu:link'] = 1;
-
-                editor.addCommand('internalLinkDialog', {
+                this.extendCkEditorDtd();
+
+                editor.addCommand('internalLinkDialog', this.getInternalLinkCommand(editor));
+                editor.addCommand('removeInternalLink', this.getRemoveInternalLinkCommand(editor));
+
+                editor.ui.addButton(
+                    'InternalLink',
+                    {
+                        label: sandbox.translate('content.ckeditor.internal-link'),
+                        command: 'internalLinkDialog',
+                        icon: '/bundles/sulucontent/img/icon_link_internal.png'
+                    }
+                );
+
+                if (editor.contextMenu) {
+                    this.addMenuSuluGroup(editor);
+
+                    editor.contextMenu.addListener(function(element) {
+                        if (element.getAscendant('sulu:link', true)) {
+                            return {
+                                internalLinkItem: CKEDITOR.TRISTATE_OFF,
+                                removeInternalLinkItem: CKEDITOR.TRISTATE_OFF
+                            };
+                        }
+                    });
+                }
+            },
+
+            addMenuSuluGroup: function(editor) {
+                editor.addMenuGroup('suluGroup');
+                editor.addMenuItem('internalLinkItem', {
+                    label: sandbox.translate('content.ckeditor.internal-link.edit'),
+                    icon: '/bundles/sulucontent/img/icon_link_internal.png',
+                    command: 'internalLinkDialog',
+                    group: 'suluGroup'
+                });
+                editor.addMenuItem('removeInternalLinkItem', {
+                    label: sandbox.translate('content.ckeditor.internal-link.remove'),
+                    icon: '/bundles/sulucontent/img/icon_remove_link_internal.png',
+                    command: 'removeInternalLink',
+                    group: 'suluGroup'
+                });
+            },
+
+            getInternalLinkCommand: function(editor) {
+                return {
                     dialogName: 'internalLinkDialog',
                     allowedContent: 'sulu:link[title,target,sulu:validation-state,!href]',
                     requiredContent: 'sulu:link[href]',
                     exec: function() {
                         var $element = $('<div/>');
-                        $('body').append($element);
+                        $('#content').append($element);
```

Comment:
Starting sandboxes outside the parents container or the body can lead to undesired bugs, because the component probably never gets destroyed. This was the case before changing this line. If you revert the change and open and close some internal link overlays in the ckeditor, you can see empty divs accumulating at the end of the body.